File: Travel/Travel.Models/OrganizedTrip/OrganizedTrip.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Travel.Models.Agency;

namespace Travel.Models.OrganizedTrip
{
    public class OrganizedTrip
    {
        public long Id { get; set; }

        public string TripName { get; set; }

        public string Destination { get; set; }

        public DateTime StartDate { get; set; }

        public DateTime EndDate { get; set; }

        public double Price { get; set; }

        public int AvailableSeats { get; set; }

        public string Description { get; set; }

        public long AgencyId { get; set; }
        public Agency.Agency Agency { get; set; }

        public string ContactInfo { get; set; }
        public  List<TripService.TripService> IncludedServices { get; set; } 
        public  List<TripTicket.TripTicket> TripTickets { get; set; } 
    }
}

File: Travel/UI/travel_mobile/lib/model/organized_trip/organized_trip.dart
import 'dart:ffi';

import 'package:json_annotation/json_annotation.dart';
import 'package:travel_mobile/model/agency/agency.dart';
import 'package:travel_mobile/model/trip_service/trip_service.dart';
import 'package:travel_mobile/model/trip_ticket/trip_ticket.dart';

part 'organized_trip.g.dart';

@JsonSerializable()
class OrganizedTripModel {
  int? id;
  int? availableSeats;
  int? numberOfSeats;
  int? agencyId;
  AgencyModel? agency;
  String destination;
  DateTime? startDate;
  DateTime? endDate;
  double? price;
  String description;
  String contactInfo;
  String tripName;
  List<TripServiceModel> includedServices;
  List<TripTicketModel> tripTickets;
  OrganizedTripModel(
      this.id,
      this.availableSeats,
      this.agency,
      this.agencyId,
      this.description,
      this.destination,
      this.startDate,
      this.endDate,
      this.contactInfo,
      this.price,
      this.includedServices,
      this.tripName,
      this.tripTickets,
      this.numberOfSeats);

  /// A necessary factory constructor for creating a new User instance
  /// from a map. Pass the map to the generated `_$UserFromJson()` constructor.
  /// The constructor is named after the source class, in this case, User.
  factory OrganizedTripModel.fromJson(Map<String, dynamic> json) =>
      _$OrganizedTripModelFromJson(json);

  /// `toJson` is the convention for a class to declare support for serialization
  /// to JSON. The implementation simply calls the private, generated
  /// helper method `_$UserToJson`.
  Map<String, dynamic> toJson() => _$OrganizedTripModelToJson(this);
}
